lib/json-schema/schema.rb

```diff
@@ -6,6 +6,8 @@ class Schema
     attr_accessor :schema, :uri, :validator
 
     def initialize(schema,uri,parent_validator=nil)
+      raise ArgumentError, "schema must be a hash; got: #{schema.inspect}" unless schema.is_a?(Hash)
```

Comment:
this change is kind of unrelated, but it results in a better error message when an array is passed as the `schema` argument (as the previous fragment-handling code was prone to do) - before the error message would end up being `TypeError: no implicit conversion of String into Integer` when trying to call `@schema['id']` below, which is rather obscure.